pylint/checkers/utils.py

```diff
@@ -802,3 +802,39 @@ def node_type(node):
     except astroid.InferenceError:
         return
     return types.pop() if types else None
+
+
+def is_singledispatch_function(node):
```

Comment:
Maybe it would be worth changing ``decorated_with`` to use a full inference loop instead of safe_infer? I am not keen on having functions tailored to a particular module, as this one.